target/scripts/startup/setup-stack.sh

```diff
@@ -1118,8 +1190,7 @@ function _setup_docker_permit
 
   if [[ -z ${CONTAINER_IP} ]]
   then
-    _notify 'err' "Detecting the container IP address failed. Check if NETWORK_INTERFACE is correctly configured."
-    _shutdown
+    _shutdown "Detecting the container IP address failed. Check if NETWORK_INTERFACE is correctly configured."
```

Comment:
Do you think, converting that to `dms_panic()` would make sense?

Reply:
Possibly.

Do you want the same/similar message or is there an appropriate generic one that could benefit from a new `PANIC_TYPE`?

Could emit the "Detecting the container IP address failed." via `_notify 'err'` prior to the shutdown/panic fatal error: "Possible misconfiguration for NETWORK_INTERFACE" (new panic type, `PANIC_MISCONFIGURED`)?

Reply:
I am not sure if I understand correctly. But the information what is misconfigured should in any case be given and not just a generic "something is misconfigured" message.

Reply:
Hopefully the new commit makes my intent more clear :)

I did it in two stages:
- 1st commit is about what was discussed here.
- 2nd commit is a refactor based on better developer experience.

Reply:
Ok, I was already a bit tired yesterday. I get now what you meant.

I understand your dms_panic approach, but I still have the feeling, that this is a bit over-engineered. There are hopefully not that much situation, where we need to "shutdown". But I don't want to be the judge here. In the end, if it works, it's okay for me 👍 
The whole time, I was more thinking of a much simpler solution (yet not so professional as yours):

Just one function `_shutdown($message)` that is called in the SSL context (you reworked in that PR + ), as well as in the three other occurrences mentioned earlier.
To keep it DRY, repeating $messages can still be put in constants:
```

readonly PANIC_NO_IP="no IP ..."
readonly PANIC_NO_HOST="Setting hostname/domainname is required'
readonly PANIC_SSL_KEY="Certificate key not found"
readonly PANIC_SSL_CERT="Certificate not found"
```

Then just call it like this:

`_shutdown($PANIC_NO_HOST)`

Yes, this is not as flexible as your solution, but do we really need that flexibility for that few "shutdown/panic" events?

Again, don't get me wrong. This is not something I say we must do it this way, just an option to discuss 😉

Reply:
> `_shutdown($PANIC_NO_HOST)`

I removed the vars so anyone that wants to panic/shutdown doesn't have to copy/paste a specific one, if they're going to lookup the options to copy/paste might as well just make it a method name and/or prevent any typo mistakes?

In my editor it also seems to autocomplete with a suggestion if I type the function name, but that might be because I've opened `helper-functions.sh` and was working on the method :sweat_smile: 

You're right that it's over-engineered for what it is and our needs atm. I wanted to panic within `_setup_tls` under some common conditions and wasn't aware of `_shutdown` at the time, so I wrote a method to reduce copy/paste if I need it in future and avoid typing similar error messages for the same thing months later :sweat_smile: 

I personally prefer the current method as it's more maintainable / flexible, than adding new string vars for more specific failures (_which has less value in co-location vs inline?_).

`_shutdown` still works directly but expects a message arg (I could make it optional if desired), which is to encourage invoking it with some context of where it was called (less of an issue with us and having a review process though :stuck_out_tongue: ).

---

I'm fine with whatever is agreed on. The current approach might be a bit much, but it works and shouldn't cause any maintenance woes, probably best to not bike shed on it too much :smile: